libvirt/resource_libvirt_domain.go

```diff
@@ -106,6 +118,34 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		domainDef.Metadata.TerraformLibvirt.Xml = metadata.(string)
 	}
 
+	if firmware, ok := d.GetOk("firmware"); ok {
+		firmwareFile, err := homedir.Expand(firmware.(string))
```

Comment:
Yes, it is a convenience function for paths that are relative to the home directory. I think it makes things easier, but I will remove it if you think it is superfluous...